src/Compilers/Server/VBCSCompilerTests/DesktopBuildServerControllerTests.cs

```diff
@@ -42,5 +45,32 @@ public void NoSetting()
                 Assert.Equal(ServerDispatcher.DefaultServerKeepAlive, _controller.GetKeepAliveTimeout());
             }
         }
+
+        [ConditionalFact(typeof(UnixLikeOnly))]
+        public void RunServerWithLongTempPath()
+        {
+            var pipeName = Guid.NewGuid().ToString("N");
+            // Make a really long path. This should work on Windows, which doesn't rely on temp path,
+            // but not on Unix, which has a max path length
+            var tempPath = new string('a', 100);
+
+            // This test fails by spinning forever. If the path is not seen as invalid, the server
```

Comment:
I assume we have a global timeout for tests?

Reply:
Yup.